packages/perseus-editor/src/widgets/interactive-graph-editor/locked-figures/locked-line-settings.tsx

```diff
@@ -167,6 +226,55 @@ const LockedLineSettings = (props: Props) => {
                 onChangeProps={(newProps) => handleChangePoint(newProps, 1)}
             />
 
+            {flags?.["mafs"]?.["locked-line-labels"] && (
+                <>
+                    <View style={styles.horizontalRule} />
+
+                    {labels.map((label, labelIndex) => (
+                        <LockedLabelSettings
+                            {...label}
+                            expanded={true}
+                            onChangeProps={(newLabel: LockedLabelType) => {
+                                handleLabelChange(newLabel, labelIndex);
+                            }}
+                            onRemove={() => {
+                                handleLabelRemove(labelIndex);
+                            }}
+                        />
+                    ))}
+
+                    <Button
+                        kind="tertiary"
+                        startIcon={plusCircle}
+                        onClick={() => {
+                            const midpoint = [
+                                ...kline.midpoint([
+                                    points[0].coord,
+                                    points[1].coord,
+                                ]),
+                            ] as Coord;
+                            // Additional vertical offset for each label so
+                            // they don't overlap.
+                            midpoint[1] -= 1 * labels.length;
```

Comment:
Why multiply by `1` here? Also, mutating the array is slightly sketch (though I see you made a copy above) - if for no other reason than because after the mutation it isn't the midpoint anymore. I might suggest using vector addition instead:

```ts
const offsetPerLabel: vec.Vector2 = [0, -1];

const labelLocation = vec.add(
  vec.scale(offsetPerLabel, labels.length),
  vec.midpoint(...),
);
  
```

Reply:
Ah, I added the `* 1` to make it more clear that that's the offset I'm using, but it a bit silly in hindsight. I like your approach way better.